In [4]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
from datetime import datetime
from sklearn.cluster import KMeans
from sklearn.decomposition import PCA
import plotly.express as px
import scipy.cluster.hierarchy as sch
from sklearn.preprocessing import PowerTransformer, StandardScaler, MinMaxScaler
from scipy.stats import kruskal
from scipy.stats import boxcox

In [5]:
df_final = pd.read_csv('data/df_final.csv')

In [6]:
df_final

Area  PIB par habitant  Population totale_x  \
0           albania       4531.032207            2873457.0   
1           algeria       4615.868744           41136546.0   
2            angola       2439.374439           30208628.0   
3           armenia       4041.995071            2851923.0   
4         australia      53954.553495           24592588.0   
..              ...               ...                  ...   
98   united kingdom      40572.121482           66058859.0   
99    united states      60322.261424          325122128.0   
100         uruguay      18995.397020            3422200.0   
101         vietnam       2992.071532           94033048.0   
102          zambia       1495.752138           17298054.0   

     Taux d'urbanisation  Taux de chômage  Taux d'inflation  \
0                 59.383           13.616          1.986661   
1                 72.052           12.000          5.591116   
2                 64.839           16.617         29.844480   
3                 63.103           12.968          0.969553   
4                 85.904            5.592          1.948647   
..                   ...              ...               ...   
98                83.143            4.330          2.557756   
99                82.058            4.355          2.130110   
100               95.240            7.890          6.218094   
101               35.213            1.874          3.520257   
102               42.976            8.518          6.577312   

     Part des énergies renouvelables  Taux de mortalité infantile  \
0                               36.9                          8.3   
1                                0.1                         21.0   
2                               52.5                         53.4   
3                               12.6                         11.7   
4                                9.9                          3.2   
..                               ...                          ...   
98                               8.8                          3.8   
99                               9.7                          5.6   
100                             60.8                          6.8   
101                             28.3                         17.2   
102                             83.5                         44.0   

     Taux d'électrification  Production_Viande de Volailles  ...  \
0                      99.9                            13.0  ...   
1                      99.5                           275.0  ...   
2                      42.9                            42.0  ...   
3                      99.7                            11.0  ...   
4                     100.0                          1269.0  ...   
..                      ...                             ...  ...   
98                    100.0                          1814.0  ...   
99                    100.0                         21914.0  ...   
100                    99.8                            33.0  ...   
101                   100.0                           918.0  ...   
102                    40.3                            49.0  ...   

     Disponibilité intérieure_Viande de Volailles  \
0                                            47.0   
1                                           277.0   
2                                           319.0   
3                                            47.0   
4                                          1171.0   
..                                            ...   
98                                         2234.0   
99                                        18266.0   
100                                          33.0   
101                                        1206.0   
102                                          60.0   

     Résidus_Viande de Volailles  Nourriture_Viande de Volailles  \
0                            0.0                            47.0   
1                            0.0                           264.0   
2                         

<div style="background-color: RGB(51,165,182);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">4 - Analyse des données</h2>
</div>

In [7]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 22 columns):
 #   Column                                                                             Non-Null Count  Dtype  
---  ------                                                                             --------------  -----  
 0   Area                                                                               103 non-null    object 
 1   PIB par habitant                                                                   103 non-null    float64
 2   Population totale_x                                                                103 non-null    float64
 3   Taux d'urbanisation                                                                103 non-null    float64
 4   Taux de chômage                                                                    103 non-null    float64
 5   Taux d'inflation                                                                   103 non-null    float64

In [8]:
df_final.describe()

PIB par habitant  Population totale_x  Taux d'urbanisation  \
count        103.000000         1.030000e+02           103.000000   
mean       17878.531236         5.663134e+07            64.609723   
std        21712.551388         1.933037e+08            20.103722   
min          484.456129         2.076300e+05            16.350000   
25%         3696.882811         4.333379e+06            54.104500   
50%         6734.475153         1.021538e+07            66.041000   
75%        25105.515660         3.879799e+07            80.898000   
max       110193.213797         1.396215e+09           100.000000   

       Taux de chômage  Taux d'inflation  Part des énergies renouvelables  \
count       103.000000        103.000000                       103.000000   
mean          7.588961          3.833840                        30.096602   
std           5.325587          4.707728                        25.162291   
min           0.564000         -0.981890                         0.000000   
25%           3.928000          1.351688                        10.650000   
50%           5.648000          2.130110                        24.100000   
75%           9.455000          4.349875                        44.850000   
max          24.830000         29.844480                        93.600000   

       Taux de mortalité infantile  Taux d'électrification  \
count                   103.000000              103.000000   
mean                     17.673786               88.922330   
std                      18.926400               21.451459   
min                       1.900000               17.600000   
25%                       3.800000               92.250000   
50%                       9.500000               99.900000   
75%                      22.700000              100.000000   
max                      87.400000              100.000000   

       Production_Viande de Volailles  \
count                      103.000000   
mean                      1019.766990   
std                       3124.475279   
min                          0.000000   
25%                         37.500000   
50%                        136.000000   
75%                        737.000000   
max                      21914.000000   

       Importations - Quantité_Viande de Volailles  ...  \
count                                   103.000000  ...   
mean                                    112.611650  ...   
std                                     197.387805  ...   
min                                       0.000000  ...   
25%                                       3.000000  ...   
50%                                      24.000000  ...   
75%                                     123.000000  ...   
max                                    1069.000000  ...   

       Disponibilité intérieure_Viande de Volailles  \
count                                    103.000000   
mean                                     947.883495   
std                                     2726.170838   
min                                        7.000000   
25%                                       60.500000   
50%                                      173.000000   
75%                                      709.000000   
max                                    18266.000000   

       Résidus_Viande de Volailles  Nourriture_Viande de Volailles  \
count                   103.000000                      103.000000   
mean                     -3.213592                      916.087379   
std                      11.979272                     2667.652479   
min                     -78.000000                        7.000000   
25%                       0.000000                       59.500000   
50%                       0.000000                      162.000000   
75%                       0.000000                      709.000000   
max                       0.000000                    18100.000000   

       Disponibilité alimentaire en quantité (kg/personne/an)_Viande de Volailles  \
count             

In [ ]:
# Création des boxplots interactifs pour chaque colonne
for col in df_final.columns:
    if col == 'Area':  # On évite de faire un boxplot pour la colonne Area elle-même
        continue
    
    # Création du boxplot interactif avec les informations sur les outliers
    fig = px.box(df_final, y=col, hover_data=['Area'], title=f"Boxplot interactif de {col}")

    # Affichage du graphique interactif
    fig.show()

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.2 - Correlation entre les données</h3>
</div>

In [ ]:
df_valeurs = df_final.select_dtypes(include=['float64', 'int64'])

In [ ]:
correlation_matrix = df_valeurs.corr(method='pearson')
plt.figure(figsize=(15, 8))
sns.heatmap(correlation_matrix, annot=True, cmap='coolwarm', vmin=-1, vmax=1)
plt.title('Matrice de Corrélation')
plt.show()

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.2.1 - Standardisation des données</h3>
</div>

In [ ]:
# Séparer les colonnes numériques et non numériques
df_pays = df_final.select_dtypes(exclude=['float64', 'int64']).columns

# Appliquer StandardScaler sur les colonnes numériques
scaler = StandardScaler()
scaled_data = scaler.fit_transform(df_final[df_valeurs])

# Convertir les données standardisées en DataFrame
scaled_df = pd.DataFrame(scaled_data, columns=df_valeurs)

# Concaténer les colonnes non numériques avec les colonnes standardisées
df_final_scaled = pd.concat([scaled_df, df_final[df_pays].reset_index(drop=True)], axis=1)

# Résultat : df_final_scaled contient les colonnes numériques standardisées et les colonnes non numériques inchangées
df_final_scaled.head()

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.2.2 - Transformation des données</h3>
</div>

In [ ]:
# Dictionnaire pour stocker les résultats
transformed_data = {}

# Sélectionner uniquement les colonnes numériques (float64, int64)
df_valeurs = df_final.select_dtypes(include=['float64', 'int64'])

# Transformation log
df_log = np.log(df_valeurs + 1)  # On applique log sur les valeurs, pas sur les noms de colonnes
transformed_data['log'] = df_log

# Box-Cox (gérer les valeurs négatives)
df_boxcox = df_valeurs.apply(lambda x: boxcox(x + 1)[0] if np.all(x > 0) else x)
transformed_data['boxcox'] = df_boxcox

# Yeo-Johnson
pt = PowerTransformer(method='yeo-johnson')
df_yeo_johnson = pd.DataFrame(pt.fit_transform(df_valeurs), columns=df_valeurs.columns)
transformed_data['yeo-johnson'] = df_yeo_johnson

# StandardScaler
scaler = StandardScaler()
df_standard = pd.DataFrame(scaler.fit_transform(df_valeurs), columns=df_valeurs.columns)
transformed_data['standard'] = df_standard

# Min-Max Scaling
scaler = MinMaxScaler()
df_minmax = pd.DataFrame(scaler.fit_transform(df_valeurs), columns=df_valeurs.columns)
transformed_data['min-max'] = df_minmax

# Vérifier les colonnes disponibles après transformation
print("Colonnes disponibles :", df_valeurs.columns)

# Visualiser les transformations
for key, data in transformed_data.items():
    plt.figure(figsize=(8, 6))
    # Utilise le bon nom de colonne, par exemple, la première colonne du DataFrame transformé
    colonne_a_visualiser = df_valeurs.columns[0]  # Remplace par une colonne valide
    plt.hist(data[colonne_a_visualiser], bins=30, alpha=0.7)
    plt.title(f'Transformation {key} - {colonne_a_visualiser}')
    plt.show()

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.2.3 - Standardisation des données</h3>
</div>

In [8]:
# Appliquer StandardScaler sur les colonnes numériques
scaler = StandardScaler()
scaled_data = scaler.fit_transform(df_yeo_johnson)

# Convertir les données standardisées en DataFrame
scaled_df = pd.DataFrame(scaled_data, columns=df_yeo_johnson.columns)

# Concaténer les colonnes non numériques avec les colonnes standardisées
df_final_scaled = pd.concat([scaled_df, df_final[df_pays].reset_index(drop=True)], axis=1)

# Résultat : df_final_scaled contient les colonnes numériques standardisées et les colonnes non numériques inchangées
df_final_scaled.head()

NameError: name 'df_yeo_johnson' is not defined

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.3 - PCA</h3>
</div>

In [ ]:
# Appliquer la PCA sur toutes les dimensions
pca = PCA().fit(scaled_data)

# Graphique du coude (variance expliquée par chaque composante)
plt.plot(range(1, len(pca.explained_variance_ratio_) + 1), pca.explained_variance_ratio_.cumsum(), marker='o')
plt.title('Graphique du coude - PCA')
plt.xlabel('Nombre de composants principaux')
plt.ylabel('Variance expliquée cumulée')
plt.show()

In [ ]:
# 2. Appliquer PCA
pca = PCA(n_components=4)  
pca_data = pca.fit_transform(scaled_data)

In [ ]:
print(pca.explained_variance_ratio_)


In [ ]:
print(pca.components_)


<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.3 - Cercle de correlation PCA</h3>
</div>

In [ ]:
# 2. Récupérer les composantes et la variance expliquée
components = pca.components_
explained_variance = pca.explained_variance_ratio_

# 3. Tracer le cercle de corrélation
plt.figure(figsize=(8, 8))

# Tracer un cercle
circle = plt.Circle((0, 0), 1, color='b', fill=False, linestyle='--')
plt.gca().add_artist(circle)

# Tracer les flèches pour chaque variable
for i, (x, y) in enumerate(zip(components[0, :], components[1, :])):
    plt.arrow(0, 0, x, y, color='r', alpha=0.7, head_width=0.05)
    plt.text(x * 1.1, y * 1.1, df_valeurs.columns[i], color='g', ha='center', va='center')

# Ajouter les étiquettes et un titre
plt.xlabel(f"PC1 ({explained_variance[0]*100:.2f}%)")
plt.ylabel(f"PC2 ({explained_variance[1]*100:.2f}%)")
plt.title("Cercle des corrélations - PCA")
plt.xlim(-1.1, 1.1)
plt.ylim(-1.1, 1.1)
plt.axhline(0, color='black', linestyle='--')
plt.axvline(0, color='black', linestyle='--')
plt.grid(True)

# Afficher le graphique
plt.show()

In [ ]:
# 1. Créer un DataFrame avec les composantes principales
components_df = pd.DataFrame(
    data=components.T,  # Transpose les composantes pour correspondre aux variables en colonnes
    index=df_valeurs.columns,  # Utilise les noms des colonnes comme index
    columns=[f"PC{i+1}" for i in range(components.shape[0])]  # Noms des composantes
)

# 2. Ajouter la variance expliquée pour chaque PC
variance_df = pd.DataFrame(
    data={'Explained Variance': explained_variance},
    index=[f"PC{i+1}" for i in range(len(explained_variance))]
)

# Afficher les DataFrames
print("Composantes principales:")
display(components_df)
print("\nVariance expliquée:")
display(variance_df)

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.4 - CAH</h3>
</div>

In [ ]:
# Créer une matrice de distances à partir des données standardisées
distance_matrix = sch.linkage(scaled_data, method='ward')

# Tracer le dendrogramme
plt.figure(figsize=(10, 7))
dendrogram = sch.dendrogram(distance_matrix)
plt.title('Dendrogramme - Classification Hiérarchique')
plt.xlabel('Pays')
plt.ylabel('Distance')
plt.show()

In [ ]:
from scipy.cluster.hierarchy import fcluster

df_final_scaled['cluster_cah'] = fcluster(distance_matrix, t=4, criterion='maxclust')  

# Visualiser les clusters obtenus
print(df_final_scaled[['Area', 'cluster_cah']])

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">4.4 - K means</h3>
</div>

In [ ]:
# Calculer l'inertie pour plusieurs valeurs de k (nombre de clusters)
inertias = []
K_range = range(1, 10)  # On teste pour 1 à 10 clusters
for k in K_range:
    kmeans = KMeans(n_clusters=k)
    kmeans.fit(pca_data)
    inertias.append(kmeans.inertia_)

# Graphique du coude (Inertie vs nombre de clusters)
plt.plot(K_range, inertias, marker='o')
plt.title('Graphique du coude - K-Means')
plt.xlabel('Nombre de clusters')
plt.ylabel('Inertie')
plt.show()


In [ ]:
# 3. Appliquer K-Means sur les données réduites
kmeans = KMeans(n_clusters=4, random_state=42)
kmeans = KMeans(n_clusters=4)
df_final_scaled['cluster_kmeans'] = kmeans.fit_predict(scaled_data)

In [ ]:
# Visualiser les clusters K-Means sur les deux premières composantes principales
plt.figure(figsize=(10, 7))
sns.scatterplot(x=pca_data[:, 0], y=pca_data[:, 1], hue=df_final_scaled['cluster_kmeans'], palette='muted', s=100)
plt.title('Clusters obtenus par K-Means')
plt.xlabel('Composante principale 1')
plt.ylabel('Composante principale 2')
plt.show()

# Visualiser les clusters CAH
plt.figure(figsize=(10, 7))
sns.scatterplot(x=pca_data[:, 0], y=pca_data[:, 1], hue=df_final_scaled['cluster_cah'], palette='muted', s=100)
plt.title('Clusters obtenus par CAH')
plt.xlabel('Composante principale 1')
plt.ylabel('Composante principale 2')
plt.show()


<div style="background-color: RGB(51,165,182);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">5 - Resultats</h2>
</div>

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.1 - K means</h3>
</div>

<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.1.1 - Analyse des variablies k means</h4>
</div>

In [ ]:
# Sélectionner les colonnes numériques
numeric_columns = df_final_scaled.select_dtypes(include=[np.number]).columns

# Grouper les données par les clusters formés par K-Means (ou CAH)
grouped_kmeans = df_final_scaled.groupby('cluster_kmeans')

# Calculer les statistiques descriptives (moyenne, écart-type, etc.) pour chaque cluster
cluster_means = grouped_kmeans[numeric_columns].mean()

# Afficher les moyennes de chaque caractéristique pour chaque cluster
print("Moyenne des caractéristiques par 'cluster_kmeans':")
display(cluster_means)

# Calculer la médiane pour chaque cluster
cluster_median = grouped_kmeans[numeric_columns].median()
print("\nMédiane des caractéristiques par 'cluster_kmeans':")
display(cluster_median)


<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.1.2 - Corr k means</h4>
</div>

In [ ]:
cluster_means_clean = cluster_means.drop(columns=['cluster_cah', 'cluster_kmeans'], errors='ignore')

plt.figure(figsize=(12, 8))
sns.heatmap(cluster_means_clean.T, annot=True, cmap="coolwarm")
plt.title("Moyenne des caractéristiques par cluster K-Means")
plt.show()

<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.1.3 - Test Kruskal</h4>
</div>

In [ ]:
# Exclure les colonnes 'cluster_cah' et 'cluster_kmeans' des colonnes numériques
filtered_columns = [col for col in numeric_columns if col not in ['cluster_cah', 'cluster_kmeans']]

In [ ]:
# Test de Kruskal-Wallis pour chaque variable numérique par cluster
kruskal_results = {}
for col in filtered_columns:
    # Groupes de données pour chaque cluster
    groups = [df_final_scaled[df_final_scaled['cluster_kmeans'] == cluster][col] for cluster in df_final_scaled['cluster_kmeans'].unique()]
    
    # Effectuer le test Kruskal-Wallis
    stat, p_value = kruskal(*groups)
    kruskal_results[col] = p_value

# Extraire et trier les p-values
sorted_p_values = sorted(kruskal_results.items(), key=lambda item: item[1])

# Afficher les 10 variables avec les p-values les plus faibles
print("Les 10 variables avec les p-values les plus faibles :")
for feature, p_value in sorted_p_values[:10]:
    print(f"{feature}: p-value = {p_value}")


In [ ]:
# Récupérer les noms des features
feature_names = [feature for feature, p_value in sorted_p_values]

# Si vous voulez seulement les 10 premières features
top_variables_kmeans = [feature for feature, p_value in sorted_p_values[:4]]

print(top_variables_kmeans)

plt.figure(figsize=(15, 10))

for i, col in enumerate(top_variables_kmeans):
    plt.subplot(3, 2, i + 1)
    sns.boxplot(x='cluster_kmeans', y=col, data=df_final_scaled, hue='cluster_kmeans' , palette="Set2")
    plt.title(f'Boxplot of {col} by K means Cluster')

plt.tight_layout()
plt.show()

<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.1.4 - Pays par cluster</h4>
</div>

In [ ]:
 # Grouper les données par 'cluster_kmeans' et lister les 'Area' (pays/régions)
grouped_areas = df_final_scaled.groupby('cluster_kmeans')['Area'].apply(lambda x: ', '.join(x))

# Convertir en DataFrame
area_cluster_df = pd.DataFrame(grouped_areas).reset_index()

# Renommer les colonnes pour plus de clarté
area_cluster_df.columns = ['Cluster K-Means', 'Areas (Pays/Régions)']

# Appliquer du style pour rendre le tableau plus présentable
styled_df = area_cluster_df.style.set_table_styles(
    [{
        'selector': 'thead th',
        'props': [('background-color', '#5DADE2'),
                  ('color', 'white'),
                  ('font-size', '14px'),
                  ('text-align', 'center')]
    },
    {
        'selector': 'tbody td',
        'props': [('background-color', '#EBF5FB'),
                  ('color', 'black'),
                  ('font-size', '12px'),
                  ('text-align', 'left'),
                  ('padding', '10px')]
    }],
    overwrite=False
).set_properties(**{
    'border': '1px solid black',
    'padding': '10px'
})

# Afficher le tableau stylisé
styled_df


<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.2 - CAH</h3>
</div>

<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.2.1 - Analyses des variablies CAH</h4>
</div>

In [ ]:
# Grouper les données par les clusters formés par K-Means (ou CAH)
grouped_CAH = df_final_scaled.groupby('cluster_cah')

# Calculer les statistiques descriptives (moyenne, écart-type, etc.) pour chaque cluster
CAH_means = grouped_CAH[numeric_columns].mean()

# Afficher les moyennes de chaque caractéristique pour chaque cluster
print("Moyenne des caractéristiques par 'cluster_cah':")
display(CAH_means)

# Calculer la médiane pour chaque cluster
CAH_median = grouped_CAH[numeric_columns].median()
print("\nMédiane des caractéristiques par 'cluster_cah':")
display(CAH_median)


<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.2.2 - Correlation</h4>
</div>

In [ ]:
CAH_means_clean = CAH_means.drop(columns=['cluster_cah', 'cluster_kmeans'], errors='ignore')

plt.figure(figsize=(12, 8))
sns.heatmap(CAH_means_clean.T, annot=True, cmap="coolwarm")
plt.title("Moyenne des caractéristiques par 'cluster_cah'")
plt.show()


<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.2.3 - Test kruskal</h4>
</div>

In [ ]:
# Test de Kruskal-Wallis pour chaque variable numérique par cluster
kruskal_results = {}
for col in filtered_columns :
    # Groupes de données pour chaque cluster
    groups = [df_final_scaled[df_final_scaled['cluster_cah'] == cluster][col] for cluster in df_final_scaled['cluster_cah'].unique()]
    
    # Effectuer le test Kruskal-Wallis
    stat, p_value = kruskal(*groups)
    kruskal_results[col] = p_value

# Extraire et trier les p-values
sorted_p_values = sorted(kruskal_results.items(), key=lambda item: item[1])

# Afficher les 10 variables avec les p-values les plus faibles
print("Les 10 variables avec les p-values les plus faibles :")
for feature, p_value in sorted_p_values[:10]:
    print(f"{feature}: p-value = {p_value}")


In [ ]:
# Récupérer les noms des features
feature_names = [feature for feature, p_value in sorted_p_values]

# Si vous voulez seulement les 10 premières features
top_variables_cah = [feature for feature, p_value in sorted_p_values[:4]]

print(top_variables_cah)



In [ ]:
plt.figure(figsize=(15, 10))

for i, col in enumerate(top_variables_cah):
    plt.subplot(3, 2, i + 1)
    sns.boxplot(x='cluster_cah', y=col, data=df_final_scaled, hue = 'cluster_cah', palette="Set2")
    plt.title(f'Boxplot of {col} by CAH Cluster')

plt.tight_layout()
plt.show()

<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color: RGB(51,165,182); ">5.2.4 - Pays par cluster</h4>
</div>

In [ ]:
# Grouper les données par 'cluster_cah' et lister les 'Area' (pays/régions)
grouped_areas = df_final_scaled.groupby('cluster_cah')['Area'].apply(lambda x: ', '.join(x))

# Convertir en DataFrame
area_cluster_df = pd.DataFrame(grouped_areas).reset_index()

# Renommer les colonnes pour plus de clarté
area_cluster_df.columns = ['Cluster cah', 'Areas (Pays/Régions)']

# Appliquer du style pour rendre le tableau plus présentable
styled_df = area_cluster_df.style.set_table_styles(
    [{
        'selector': 'thead th',
        'props': [('background-color', '#5DADE2'),
                  ('color', 'white'),
                  ('font-size', '14px'),
                  ('text-align', 'center')]
    },
    {
        'selector': 'tbody td',
        'props': [('background-color', '#EBF5FB'),
                  ('color', 'black'),
                  ('font-size', '12px'),
                  ('text-align', 'left'),
                  ('padding', '10px')]
    }],
    overwrite=False
).set_properties(**{
    'border': '1px solid black',
    'padding': '10px'
})

# Afficher le tableau stylisé
styled_df


<div style="background-color: RGB(51,165,182);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">6 - Conclusion et recommandations</h2>
</div>

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color:; ">6.1 - Analyse des Clusters pour l'Exportation de Poulets Biologiques</h3>
</div>

<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color:; ">6.1.1 - Clusters K-means</h4>
</div>

##### Cluster 0 (pays à faible développement) :
- **Caractéristiques** : Faible taux d'électrification, mortalité infantile élevée, consommation et production de volaille très faibles.
- **Exemple de pays** : Afghanistan, Angola, Bangladesh.
- **Profil** : Ces pays manquent d'infrastructures et de production alimentaire avancée. La demande en produits biologiques est faible, mais il pourrait y avoir une opportunité à long terme à mesure que ces infrastructures s'améliorent.

##### Cluster 1 (pays développés avec une forte consommation) :
- **Caractéristiques** : Population élevée, forte production et consommation de volaille et d'œufs, taux d'électrification et accès à Internet très élevés.
- **Exemple de pays** : Argentine, Brésil, Chine, États-Unis.
- **Profil** : Ce groupe comprend les principaux producteurs et consommateurs de volaille dans le monde. Les infrastructures sont solides, et il y a une forte demande pour des produits de meilleure qualité, comme les produits biologiques.

##### Cluster 2 (petits pays insulaires ou pays à économie modérée) :
- **Caractéristiques** : PIB modéré à élevé, population et taux de mortalité infantile faibles, infrastructures avancées.
- **Exemple de pays** : Antigua-et-Barbuda, Bahamas, Barbade.
- **Profil** : Ce cluster regroupe des pays plus petits et plus riches avec une consommation alimentaire plus niche. Le marché biologique pourrait être une opportunité pour ces pays, mais la taille du marché reste limitée.

##### Cluster 3 (pays en développement rapide) :
- **Caractéristiques** : Forte urbanisation, consommation modérée de volaille, infrastructures en développement rapide.
- **Exemple de pays** : Albanie, Algérie, Australie.
- **Profil** : Ces pays sont en phase de transition vers des infrastructures plus modernes et une urbanisation rapide. Le marché pour les produits biologiques pourrait croître à mesure que la classe moyenne se développe.



<div style="border: 1px solid RGB(51,165,182);" >
<h4 style="margin: auto; padding: 20px; color:; ">6.1.2 - Clusters CAH</h4>
</div>

##### Cluster 1 (grands producteurs avec forte consommation de volaille) :
- **Caractéristiques** : Consommation et production élevées de volaille et d'œufs, infrastructures développées (électrification, Internet).
- **Exemple de pays** : Brésil, États-Unis, Chine.
- **Profil** : Les pays de ce groupe sont des cibles immédiates pour l'exportation de poulets biologiques, car la demande pour des produits de qualité y est déjà présente.

##### Cluster 2 (pays à infrastructures faibles) :
- **Caractéristiques** : Faible électrification, mortalité infantile élevée, production et consommation de volaille très faibles.
- **Exemple de pays** : République Centrafricaine, Djibouti, Haïti.
- **Profil** : Ce groupe comprend des pays ayant des infrastructures limitées. Ce ne sont pas des cibles immédiates, mais ils pourraient présenter une opportunité à long terme avec des améliorations socio-économiques.

##### Cluster 3 (pays développés avec consommation modérée) :
- **Caractéristiques** : PIB par habitant élevé, forte urbanisation, production modérée de volaille.
- **Exemple de pays** : France, Allemagne, Japon.
- **Profil** : Ces pays ont des infrastructures développées et une demande pour des produits biologiques de qualité, mais leur production interne de volaille est relativement modeste.

##### Cluster 4 (pays émergents à développement rapide) :
- **Caractéristiques** : Fort taux de croissance, infrastructures en développement, production et consommation croissantes de volaille.
- **Exemple de pays** : Algérie, Indonésie, Inde.
- **Profil** : Ces pays sont en plein développement et présentent une opportunité à moyen terme. Leur marché pour les produits biologiques pourrait se renforcer à mesure que leurs infrastructures s'améliorent.

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color:; ">6.2 - Conclusion globale</h3>
</div>

L'étude des clusters a révélé des différences marquées entre les pays en termes de production et de consommation de volaille, d'infrastructures, et de développement économique. Les pays les plus favorables pour l'exportation de poulets biologiques sont ceux qui présentent une consommation élevée, une population importante, et des infrastructures solides. D'autres pays en développement rapide pourraient devenir des cibles intéressantes à moyen terme.

<div style="border: 1px solid RGB(51,165,182);" >
<h3 style="margin: auto; padding: 20px; color:; ">6.3 - Recommandations stratégiques</h3>
</div>

- **Priorité immédiate (Cluster 1 K-means et CAH)** :
  - Cible les grands marchés comme la Chine, le Brésil, et les États-Unis. Ces pays sont déjà de gros consommateurs de volaille, et la demande pour des produits biologiques est en pleine expansion.

- **Opportunités à moyen terme (Cluster 3 K-means et CAH)** :
  - Les pays émergents avec une urbanisation rapide et une classe moyenne en croissance, comme l'Inde et l'Indonésie, sont des cibles à surveiller. L'amélioration des infrastructures dans ces pays ouvrira des opportunités pour des produits biologiques.

- **Opportunités à long terme (Cluster 2 CAH)** :
  - Les pays avec des infrastructures faibles ne sont pas des cibles immédiates, mais à long terme, ils pourraient devenir intéressants à mesure que leurs économies se développent.